pdns/pdns_recursor.cc

```diff
@@ -3555,6 +3556,13 @@ try
         last_carbon = g_now.tv_sec;
       }
     }
+    if (worker && t_pdl != nullptr) {
```

Comment:
`worker` is true for the distributor thread, which means we will also call the `maintenance` function from this thread which is probably useless. It doesn't matter much though.